constructor/structures.py

```diff
@@ -0,0 +1,46 @@
+import enum
+import io
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import OrdinalEncoder
+import torch
+
+
+class TaskType(enum.Enum):
+    trainClassification = 0
+    trainRegression = 1
+    applyToData = 2
+
+
+class Task(object):
+    def __init__(self, args):
+        self.epoch = args[2]
+        self.task_type = TaskType(args[5])
+
+        self.dataset = args[7]
+        self.nnDescription = args[8]
+        self.model = args[9]
+
+
+def prepare_data(df):
+    df = df.dropna()
+    ord_enc = OrdinalEncoder()
+    col = df.select_dtypes(include=['object']).columns
+    df[col] = ord_enc.fit_transform(df[col])
+
+
+class Dataset(object):
+    def __init__(self, args):
+        data_bytes, target_name = args[1:]
+        f = io.BytesIO(data_bytes)
+        df = pd.read_csv(f)
```

Comment:
In general, you shouldn't load the whole ds into memory at once, as it may be too huge. Use batching and stochastic gradient descent.